### ISSUE TITLE
TextVectorizationLayer supporting custom padding token

### ISSUE THREAD

**Issue by EduardoDixo** [tensorflow/tensorflow#38043]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): v2.2.0-rc2
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

When using the `tf.keras.layers.experimental.preprocessing.TextVectorization` we are not able to specify a padding token which should map to the value 0. When we have a sequence such as:

```
tf.constant(['<PAD> word1 word2 OOV <EOS>'], dtype=tf.string)
```
in which the `<PAD>` is the padding token. If the TextVectorization  `output_sequence_length` is bigger than the size of the previous sequence (5), for example 10, we would get:

```
vectorize_layer(tf.expand_dims(tf.constant(['<PAD> word1 word2 OOV <EOS>'], dtype=tf.string), axis=0))

`<tf.Tensor: shape=(1, 10), dtype=int64, numpy=array([[5,   2,   3, 1,  6,   0,   0,   0,   0,   0]])>`
```
(the oov token maps to 1, while the `<PAD>` token has a value different than 0 which is the default padding value by the layer). In my preprocessing I am generating sequences of the same size in each batch and the `<PAD>` token only appears at the beginning, so the model is oblivious to the 0 values and only sees the 5's as the padding token ID. However, it would be nice if we only had one token ID for the padding.

I overrided this behavior in tf v2.1.0 by doing the following:

```
vectorize_layer.adapt(tf.data.TextLineDataset('train.txt').batch(32))
pad_token_vocab = vectorize_layer._convert_to_ndarray(['<PAD>'])
pad_token_value = vectorize_layer._convert_to_ndarray([0])
vectorize_layer._insert_table_data(pad_token_vocab, pad_token_value)
vectorize_layer.set_vocabulary[('<EOS>'], append=True)
```
However, there were major changes from v2.1.0 to v.2.2-rc2 so this is no longer possible. 

**Will this change the current api? How?** Adding an optional argument: padding_token.

**Who will benefit with this feature?** I believe it's a nice to have for everyone training language models.

**Any Other info.**

**Comment by mohantym** [comment]
Hi @EduardoDixo! 
We are checking to see whether you still need help in this issue . Have you tried latest version (TF 2.7) yet?
